specs/Deployment/DeploymentAPI.md

```diff
@@ -246,12 +264,164 @@ Here's the previous example extended with the `Show UI on error` option:
     }
 ```
 
+## Repair
+
+This API will always attempt to repair WindowsAppRuntime regardless of it's state. For that reason,
+this API should be called only when deployment status is not OK to avoid an unecessary repair even
+when WindowsAppRuntime is in good state. All information about the version, channel, and
+architecture needed are derived from the current WinAppSDK Framework package.
+
+Since both the Main _and_ the Singleton packages will be repaired, it is possible that Main package
+repair has failed and Initialize returns with PackageRepairFailed status (OR) it is possible that
+Main package is repaired successfully but the Singleton package repair has failed and Initialize
+returns with repair has failed and Initialize returns with PackageRepairFailed status (OR) it is
+possible that status. In both cases, the returned WindowsAppRuntimeStatus will contain the error of
+first package repair failure. There will be no rollback of any successfully repaired packages. See
+the
+[developer docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/deploy-packaged-apps#address-installation-errors)
+for instructions on how to handle errors.
+
+```C#
+    if (DeploymentManager.GetStatus().Status == DeploymentStatus.PackageRepairRequired ||
+        DeploymentManager.GetStatus().Status == DeploymentStatus.Unknown)
```

Comment:
Shouldn't this really be "if status != DeploymentStatus.PackageRepairFailed"?

We want to attempt a repair for all errors, except the error that says "Don't try to repair; it won't work."

Otherwise, if we add a new status like "PackageRepairRecommended" or "PreviousRepairTimedOut", we will fail to attempt a repair.